model/DataList.php

```diff
@@ -503,7 +503,7 @@ function(DataQuery $query, DataList $list) use ($field, &$columnName, $linearOnl
 				$fieldName = array_pop($relations);
 
 				// Apply
-				$relationModelName = $query->applyRelation($field, $linearOnly);
+				$relationModelName = $query->applyRelation($relations, $linearOnly);
```

Comment:
This previously added the $field to the relation, but relied on it always forgiving the extra field and ignoring it.